server/text_generation_server/models/causal_lm.py

```diff
@@ -719,18 +531,19 @@ def forward(
 
     @tracer.start_as_current_span("generate_token")
     def generate_token(self, batch: CausalLMBatch) -> Tuple[List[Generation], Optional[CausalLMBatch]]:
+        trace(f'GENERATE ({len(batch.requests)}){[r.data.id for r in batch.requests]}, {batch.input_ids.shape}')
         self.step = self.step + 1
         if self.hb_profer_started == True and self.step > self.profiling_warmup_steps + self.profiling_steps:
             self.hb_profer.stop()
             self.hb_profer_started = False
 
         if self.is_optimized_for_gaudi:
-            token_idx = torch.tensor(batch.attention_mask.shape[-1] - batch.padding_right_offset).to(self.device)
+            token_idx = torch.tensor(batch.attention_mask.shape[-1] - batch.right_padding).to(self.device)
             attention_mask = batch.attention_mask
-
         else:
             token_idx = None
             # slice the attention mask to the correct shape
+            # TODO fix me!
```

Comment:
Is it something to fix in this PR?

Reply:
No. This has a low priority, as we are aiming in is_optimized_for_gaudi case.
Summing up, I know there are a few TODOs in this code, but these are mostly related with future work.